### PR TITLE
refactor: refactor PreMountLoader

### PR DESCRIPTION
**Pull request by seekshiva**
## Type of Change

- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement
- [x] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [ ] CI/CD

## Checklist

- [x] I ran `npm run re:build`
- [ ] I reviewed submitted code
- [ ] I added unit tests for my changes where possible

### CODE REVIEW

File: src/Payments/PreMountLoader.res [hunk 1]
```diff
@@ -1,165 +1,129 @@
...
-      switch key {
-      | "payment_methods" => setPaymentMethodsResponseSent(_ => true)
-      | "session_tokens" => setSessionTokensResponseSent(_ => true)
-      | "customer_payment_methods" => setCustomerPaymentMethodsResponseSent(_ => true)
-      | "saved_payment_methods" => setSavedPaymentMethodsResponseSent(_ => true)
-      | _ => ()
+      let messageHandler = (ev: Window.event) => {
+        let json = ev.data->Utils.safeParse
+        let dict = json->Utils.getDictFromJson
+        if dict->Dict.get("sendSavedPaymentMethodsResponse")->Belt.Option.isSome {
```
Comment (PritishBudhiraja) [inline src/Payments/PreMountLoader.res:93]:
```suggestion
        if dict->Dict.get("sendSavedPaymentMethodsResponse")->Option.isSome {
```

File: src/Payments/PreMountLoader.res [hunk 2]
```diff
@@ -1,165 +1,129 @@
...
-    ~sessionId,
-    ~source=Loader,
-    ~merchantId=publishableKey,
-    ~clientSecret,
-  )
+let sendPromiseData = (promise, key) => {
+  open Promise
+  promise
+  ->then(res => resolve(res))
+  ->catch(_err => resolve(JSON.Encode.null))
```
Comment (PritishBudhiraja) [inline src/Payments/PreMountLoader.res:5]:
```suggestion
  ->catch(_ => resolve(JSON.Encode.null))
```

File: src/Payments/PreMountLoader.res (hunk 1 above)
Comment (PritishBudhiraja) [reply src/Payments/PreMountLoader.res]: Can we have 
```
let isKeyPresentInDict = (dict, key) => dict->Dict.get(key)->Option.isSome
```

this in Utils and use that everywhere as this is been called multiple times

**Review by PritishBudhiraja — APPROVED** [review]

**Review by sakksham7 — APPROVED** [review]
tested locally apple flow, normal card flow working fine

**Pull request merged**